mail.gemspec

```diff
@@ -14,6 +14,9 @@ Gem::Specification.new do |s|
   s.rdoc_options << '--exclude' << 'lib/mail/values/unicode_tables.dat'
 
   s.add_dependency('mini_mime', '>= 0.1.1')
+  s.add_dependency('net-smtp')
+  s.add_dependency('net-imap')
+  s.add_dependency('net-pop')
```

Comment:
Do these gems no-op on older Rubies (1.8, 1.9, etc)?

Reply:
These gems yes, however they themselves have `digest` as dependency which is causing some doable loading problem:

```
/Users/byroot/.gem/ruby/2.7.2/gems/digest-3.0.0/lib/digest.rb:6: warning: already initialized constant Digest::REQUIRE_MUTEX
/opt/rubies/2.7.2/lib/ruby/2.7.0/digest.rb:6: warning: previous definition of REQUIRE_MUTEX was here
....
```

I already submitted the fix in bundler https://github.com/rubygems/rubygems/pull/4989, hopefully it should be fixed soon. I totally understand if you want to wait for bundler to fix this first.

Reply:
I think this is already fixed @jeremy @casperisfine. Can we release this?

Reply:
Would it make sense to wrap these in an if statement that is only true for Ruby 3.1+?

Reply:
The problem here is those gems can't be installed on older versions, @casperisfine's reply only shows part of the issue.

For example on 2.0.0:
```
$ gem i net-smtp            
Fetching: timeout-0.2.0.gem (100%)
Successfully installed timeout-0.2.0
Fetching: io-wait-0.2.1.gem (100%)
ERROR:  Error installing net-smtp:
	io-wait requires Ruby version >= 2.6.0.

$ gem i net-imap
ERROR:  Error installing net-imap:
	io-wait requires Ruby version >= 2.6.0.

$ gem i net-pop 
ERROR:  Error installing net-pop:
	io-wait requires Ruby version >= 2.6.0.
```

Reply:
@nertzy  I am pretty sure that an if statement in a .gemspec ends up getting evaluated at *gem release time*, and trying to condition on ruby version would end up being on what ruby version is running in the *environment doing the gem release*, and then fixed in the release. 

My understanding is you can't actually do dynamic conditions based on the running environment in a gemspec like that.

Reply:
Ideally those gems ('net-smtp', 'net-imap' and 'net-pop') would all NOOP when installed on an older unsupported Ruby.  That would just solve it for everyone.

Reply:
So I'm still trying to find solutions upstream, but ultimately would you be open to drop pre 2.5 support to get this moved forward now?

Reply:
ruby-core doesn't seem generally happy to need to explicitly support (even as noop) such old and EOL-since-years Ruby version based on lots of recent discussions, which brings the age-old question: do gems depending on net-* and specifically `mail` still need to support Ruby <= 2.5 today?